src/System/Hooks/useCollectorSignals.tsx

```diff
@@ -0,0 +1,183 @@
+// import { useCallback } from "react"
+import { graphql, useFragment } from "react-relay"
+import {
+  useCollectorSignals_me$key,
+  useCollectorSignals_me$data,
+} from "__generated__/useCollectorSignals_me.graphql"
+import { useCollectorSignals_artworksConnection$key } from "__generated__/useCollectorSignals_artworksConnection.graphql"
+import { useCollectorSignals_artwork$key } from "__generated__/useCollectorSignals_artwork.graphql"
+import { extractNodes } from "Utils/extractNodes"
+import { useGlobalMe } from "System/Hooks/useGlobalMe"
+import { useFeatureFlag } from "System/Hooks/useFeatureFlag"
+import React, { createContext, useContext, useEffect, useMemo } from "react"
+
+interface SignalResult {
+  partnerOffer?: any // todo
+}
+
+type UseCollectorSignalsResult = Record<string, SignalResult>
+
+interface CollectorSignalsContextProps {
+  signals: UseCollectorSignalsResult
+}
+
+const CollectorSignalsContext = createContext<CollectorSignalsContextProps>({
+  signals: {},
+})
+
+const providerFactory = (signals: UseCollectorSignalsResult) => {
+  return ({ children }) => (
+    <CollectorSignalsContext.Provider value={{ signals }}>
+      {children}
+    </CollectorSignalsContext.Provider>
+  )
+}
+
+/**
+ * Hook to access the collector signals context. Must be used below
+ * the provider returned by `useLoadCollectorSignals()`
+ */
+export const useCollectorSignals = (artworkID: string): SignalResult => {
+  const signalsMap = useContext(CollectorSignalsContext)
+  console.log("***", { signalsMap, artworkID })
+  return signalsMap.signals[artworkID] ?? {}
+}
+
+/**
+ * Hook to load and process collector signals for a list of artworks
+ */
+export const useLoadCollectorSignals = ({
+  artworks,
+}: {
+  artworks?: useCollectorSignals_artworksConnection$key | null
+}): {
+  signals: UseCollectorSignalsResult
+  CollectorSignalsProvider: React.FC
+} => {
+  const { me } = useGlobalMe()
+  const meData = useFragment<useCollectorSignals_me$key>(ME_FRAGMENT, me)
+
+  const artworksData = useFragment(ARTWORKS_CONNECTION_FRAGMENT, artworks)
+  const partnerOffersEnabled = useFeatureFlag("emerald_signals-partner-offers")
+
+  const artworksDataNodes = extractNodes(artworksData)
+
+  const artworkIDs = JSON.stringify(
+    artworksDataNodes.map(artwork => artwork.internalID)
+  )
+
+  useEffect(() => {
+    console.log("*** First render of useLoadCollectorSignals")
+  }, [])
+
+  const signals = useMemo(() => {
```

Comment:
As noted in slack, will update to use 1:1 naming for variables (ie `collectorSignals`)